samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -490,6 +455,17 @@ protected void setSerde(Log4jSystemConfig log4jSystemConfig, String systemName)
     }
   }
 
+  /**
+   * If the partition count was explicitly specified, then use that. Otherwise, use the container count as the partition
+   * count.
+   */
+  private int calculateStreamPartitionCount(Config config) {
```

Comment:
`getPartitionCount` assumes that the instance variable `config` is available, but `config` isn't actually guaranteed to be available until `LoggingContextHolder` has a config filled in, so this could result in the caller accidentally using `getPartitionCount` too early. `calculateStreamPartitionCount` accepts `config` as an argument, which forces the callers of that method to ensure `config` is available.
I didn't want to change the `getPartitionCount` signature just in case log4j/log4j2 still needed the public getter. Also, based on the javadoc for `getPartitionCount`, it is tied to some log4j/log4j2 configuration, so it seemed like it would be reasonable to have it only consider the configured value.

Reply:
since this is also a breaking change, i want to dig a bit more into this.
so earlier `getPartitioncount` used getConfig() which fetched the config when config became available (different conditions for AM vs container). So even with this earlier flow, if getPartitionCount was ever called before config became available then there would be a problem. caller of getPartitionCount has already ensured that this call is made only after config becomes available. 

i am seeing LoggingContextHolder as a way to avoid the separate checks for AM and container and even avoid calling them isAM and make it cleaner. it is not going to make config available sooner than before. given this, its more a refactor for cleaner and readable code than logic changes. Hence, i am uncomfortable making breaking changes and would like to avoid if possible.

Reply:
Good point that callers of `getPartitionCount` already needed to make sure config was available. I will change this back.